### ISSUE TITLE
Package install fails inside vscode devcontainer

### ISSUE THREAD

**Issue by imbuedhope** [jetperch/joulescope_driver#2]
I have a [vscode devcontainer](https://code.visualstudio.com/docs/devcontainers/containers) set up for work on a project that depends on the `joulescope` library and noticed that my container build breaks when trying to install this package.

I am a bit confused about what's going on — I tested the install on a fresh venv in macOS and works as expected. Installing requests first does get it to work so I've hotfixed it that way for now.

I can share the Dockerfile if that would help but it's just `mcr.microsoft.com/vscode/devcontainers/python:0-3.10-bullseye` (which is Debian 11 + vscode container packages + python3.10) with `libudev-dev` installed with `apt-get`.

```
vscode ➜ /workspaces/<redacted project name> (<redacted branch name> ✗) $ pip install pyjoulescope_driver
Defaulting to user installation because normal site-packages is not writeable
Collecting pyjoulescope_driver
  Using cached pyjoulescope_driver-1.1.0.tar.gz (2.6 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-_eygg_74/pyjoulescope-driver_6539fe89b7a442d0b48ca43748203a08/setup.py", line 32, in <module>
          from pyjoulescope_driver.release import releases_get_from_network
        File "/tmp/pip-install-_eygg_74/pyjoulescope-driver_6539fe89b7a442d0b48ca43748203a08/pyjoulescope_driver/release.py", line 30, in <module>
          import requests
      ModuleNotFoundError: No module named 'requests'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

[notice] A new release of pip available: 22.2.2 -> 22.3.1
[notice] To update, run: pip install --upgrade pip
```

**Comment by mliberty1** [comment]
Hi @imbuedhope - Thanks for the report.  The problem is that pip3 does not install the `requests` module.  See [this forum post](https://forum.joulescope.com/t/joulescope-software-for-js220/506).  The temporary solution is to run this command before `pip3 install -U joulescope_driver`:

    pip3 install -U numpy Cython request setuptools wheel

We actually have fix in progress, which should release early next week.

**Comment by mliberty1** [comment]
Fixed by 1.1.1.  `pip3 install -U pyjoulescope_driver` should now work for you:

```console
mliberty@taurus:~$ python3 -VV
Python 3.10.6 (main, Nov  2 2022, 18:53:38) [GCC 11.3.0]
mliberty@taurus:~$ python3 -m venv ~/venv/tmp
mliberty@taurus:~$ . ~/venv/tmp/bin/activate
(tmp) mliberty@taurus:~$ pip3 install -U pyjoulescope_driver
Collecting pyjoulescope_driver
  Downloading pyjoulescope_driver-1.1.1.tar.gz (2.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.6/2.6 MB 38.8 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting requests
  Using cached requests-2.28.1-py3-none-any.whl (62 kB)
Collecting numpy>=1.20
  Using cached numpy-1.23.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.1 MB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.9.24-py3-none-any.whl (161 kB)
Collecting charset-normalizer<3,>=2
  Using cached charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.13-py2.py3-none-any.whl (140 kB)
Building wheels for collected packages: pyjoulescope_driver
  Building wheel for pyjoulescope_driver (pyproject.toml) ... done
  Created wheel for pyjoulescope_driver: filename=pyjoulescope_driver-1.1.1-cp310-cp310-linux_x86_64.whl size=2991934 sha256=0886174ab4b7218c8224f266fb1808149469355b1c4d8d8fc74e292cb836a9b6
  Stored in directory: /home/mliberty/.cache/pip/wheels/a0/5a/a3/16bf581067f05969550611f7a44a9bee351fb6d69a389063bb
Successfully built pyjoulescope_driver
Installing collected packages: urllib3, numpy, idna, charset-normalizer, certifi, requests, pyjoulescope_driver
Successfully installed certifi-2022.9.24 charset-normalizer-2.1.1 idna-3.4 numpy-1.23.5 pyjoulescope_driver-1.1.1 requests-2.28.1 urllib3-1.26.13
(tmp) mliberty@taurus:~$
```

If you still see an issue, please reopen this issue.
